### PR TITLE
use `ssh -o BatchMode=yes` when making SSH connections

### PR DESCRIPTION
**Pull request by terencehonles**
SSH connections are now run with `-o BatchMode=yes` to prevent interactive prompts which may tie up a step until it times out. From the SSH docs:

>  If set to yes, user interaction such as password prompts and host key confirmation requests will be disabled. This option is useful in scripts and other batch jobs where no user is present to interact with ssh.

This will allow a step to immediately fail and the error can now be inspected. Steps which make their own SSH connections are encouraged to also use `-o BatchMode=yes` in order to not have surprising behaviors if the host configuration is different.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by terencehonles** [comment]
I'm not sure if there are any implementations that do not support this flag, but I ran into this by my own SSH connection that didn't have a host key set up in the configuration I was running, and it wasn't obvious what was happening since there were other things acting strange (#5848). Once I got to the bottom of it I decided to look at the Git step to see how it worked, and it just happened that I had already accepted GitHub's host key and removing it caused the step to hang.

File: master/buildbot/www/ldapuserinfo.py [hunk 1]
```diff
@@ -26,15 +26,19 @@
...
-
 from twisted.internet import threads
 
 from buildbot.util import bytes2unicode
 from buildbot.util import flatten
 from buildbot.www import auth
 from buildbot.www import avatar
 
+try:
+    import ldap3
```
Comment (p12tic) [inline master/buildbot/www/ldapuserinfo.py:37]: This change needs to be split off to a separate PR.

**Review by p12tic — CHANGES_REQUESTED** [review]
Looks good, but please split off the ldap3 import change to a separate commit/PR.

File: master/buildbot/www/ldapuserinfo.py (hunk 1 above)
Comment (terencehonles) [reply master/buildbot/www/ldapuserinfo.py:37]: Ok, I only included it because I noticed `make trial` was failing on this test because just importing this file would cause the test suite to fail.

Comment (terencehonles) [reply master/buildbot/www/ldapuserinfo.py]: Moved to #5857

**Review by p12tic — APPROVED** [review]

File: master/buildbot/newsfragments/ssh-batch-mode.feature [hunk 2]
```diff
@@ -0,0 +1,11 @@
+SSH connections are now run with ``-o BatchMode=yes`` to prevent interactive
```
Comment (p12tic) [inline master/buildbot/newsfragments/ssh-batch-mode.feature:1]: For the future, it's best to keep the release notes short, one-two sentences unless it's a change that affects majority of users in a way that requires their attention (e.g. deprecation/replacement of frequently used feature). It's best to explain this in the documentation of affected steps.
